week-15/quizzes/03-reducers.md

```diff
@@ -0,0 +1,92 @@
+# Reducers Quiz
+
+This quiz is to test you on what reducers are and how to use them.
+
+## The questions
+
+<quiz>
+  <question>
+    <p>
+      The purpose of a reducer function is to:
+    </p>
+    <answer>Reduce the `state` object to a single value</answer>
+    <answer correct>Update `state`</answer>
+    <answer>Make a new `state` object</answer>
+    <answer>Dispatch an `action`</answer>
+    <explanation>
+      The Redux store has a reducer function for updating the current state and 
+      returning that new state. 
+    </explanation>
+  </question>
+</quiz>
+<quiz>
+  <question multiple>
+    <p>
+      A reducer accepts which of the following parameters? 
+    </p>
+    <answer>Initial `state`</answer>
+    <answer>Payload</answer>
+    <answer correct>Current `state`</answer>
+    <answer>Previous `state`</answer>
+    <answer>Dispatch</answer>
+    <answer correct>Action</answer>
+    <explanation>
+      The reducer function receives the current state and an action, updates the
+      state appropriately based on the action.type. 
+    </explanation>
+  </question>
+</quiz>
+<quiz>
+  <question>
+    <p>
+      A reducer should mutate the current `state` and return the same object.
+    </p>
+    <answer correct>false</answer>
+    <answer>true</answer>
+    <explanation>
+      False, a Redux reducer is a pure function so you must never mutate its 
+      arguments (i.e. `state` and `action`). Your reducer must return a new 
+      object if the `state` changes.
+    </explanation>
+  </question>
+</quiz>
+<quiz>
+  <question>
+    <p>
+      In an application with muliple reducers, what does the `state` parameter 
+      of a single reducer coorespond to? 
+    </p>
+    <answer>The entire application state</answer>
+    <answer>A new state object to populate</answer>
+    <answer correct>The single slice of state that specific reducer manages
+    </answer>
+    <answer>There is no state parameter when utilizing multiple reducers
+    </answer>
+    <explanation>
+      A reducer can be split up into multiple reducers handling separate,
+      independent _slices_ of state. Because each reducer only handles a single 
+      slice of state, its `state` parameter corresponds only to the part of 
+      the state that it manages and it only responds to actions that concern 
+      that slice of state.
```

Comment:
A reducer for another slice of state can respond to actions from another slice of state if the `action.type` matches a case in that reducer

Reply:
Right- do you think I can add that to the explanation without making it too confusing? Or do you think that breaks the question? Most of that explanation was pulled from the reading.

Reply:
I think putting that in the explanation would be alright, even if it does look a little long.

Reply:
Working to put a better explanation together, but I don't want to counter what the reading has. Should we consider updating that as well? 

It currently says “it only responds to actions that concern that slice of state”. 

If we don't I was thinking about adding something like this: 
"A reducer can be split up into multiple reducers handling separate, independent _slices_ of state. Because each reducer only handles a single slice of state, its `state` parameter corresponds only to the part of the state that it manages and it only responds to actions that concern that slice of state. It is possible for actions from another slice of state to hit another reducer if the `action.type` matches a case in that reducer. Because of this, it is best practice to make descriptive action.type strings that could only match one reducer case.”

Reply:
We can definitely update the reading to be a little more accurate. I'd prefer that to make both the explanation here and the reading follow the same kind of idea

Reply:
Sounds good. Does that explanation look alright?

Reply:
Yup, that looks good to me!